src/coreclr/jit/optcse.cpp

```diff
@@ -3318,6 +3323,21 @@ class CSE_Heuristic
             bool doCSE = PromotionCheck(&candidate);
 
 #ifdef DEBUG
+
+            if (doCSE)
+            {
+                const int attempt = m_pCompiler->optCSEattempt++;
+
+                if (m_pCompiler->info.compMethodHash() == (unsigned)JitConfig.JitCSEHash())
+                {
+                    doCSE = ((1ULL << attempt) & ((unsigned long long)JitConfig.JitCSEMask())) != 0;
```

Comment:
Should you check that `attempt` is in range? i.e.,
```
if (attempt >= 32) { // or `sizeof(unsigned int) * BITS_PER_BYTE`
  doCSE = false;
} else {
...
}
```

Presumably JitCSEMask is a 32-bit int.